central/usage/csv/csv_handler.go

```diff
@@ -0,0 +1,74 @@
+package usagecsv
+
+import (
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/usage/datastore"
+	"github.com/stackrox/rox/pkg/protoconv"
+)
+
+var bom = ([]byte)("\uFEFF")
```

Comment:
> The time format it uses in the helper function is not a good fit for csv.
What's not good about it? If you've found an issue, could you please file a bug or leave a comment in `pkg/csv` to warn the next user?

It is unfortunate to introduce yet another ad-hoc csv writer here given you borrowed quite a few "good" points. Since you're writing a bunch of code for csv handling anyway, why not doing it in `pkg/csv` package for everyone else to use? Streaming writer looks like a good idea to me.

Reply:
>> The time format it uses in the helper function is not a good fit for csv.

> What's not good about it?

I find `time.RFC1123` layout used in `pkg/csv/utils` not the best fit for CSV, because:
1. It uses English abbreviations for days and months;
2. It uses comma for the day-date separation, so the value requires double quotes;
3. Comment in [time/format.go](https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/time/format.go;l=27-32):
    ```go
    // Note that the RFC822, RFC850, and RFC1123 formats should be applied
    // only to local times. Applying them to UTC times will use "UTC" as the
    // time zone abbreviation, while strictly speaking those RFCs require the
    // use of "GMT" in that case.

    // In general RFC1123Z should be used instead of RFC1123 for servers
    // that insist on that format, and RFC3339 should be preferred for new protocols.
    ```

Reply:
4. There is a bug in `pkg/csv/writer.go`: when adding UTF-8 BOM, we add it to the first header value instead of writing to the stream. This causes BOM to be wrapper in quotes if the first value needs to be quoted. (Demo: https://play.golang.com/p/8Q34s_jlsD-)